slothy/helper.py

```diff
@@ -214,7 +235,9 @@ def reduce_source(src):
         assert SourceLine.is_source(src)
         for l in src:
             l.reduce()
-        return [ l for l in src if l.has_text() and not AsmHelper.is_alignment_directive(l) ]
+        return [ l for l in src if l.has_text() and
+                 not AsmHelper.is_alignment_directive(l) and
```

Comment:
I think the reason why I originally omitted this was that allocation directives could be used outside of the optimized region as well -- in which case the optimized source file would no longer be functional.

Reply:
Any chance we can keep this behaviour, and move directives out of optimization regions in the example files?